internal/util/setters/setters.go

```diff
@@ -158,43 +178,68 @@ func PerformSetters(path string) error {
 			if err != nil {
 				return err
 			}
-
-			err = kio.Pipeline{
-				Inputs: []kio.Reader{rw},
-				Filters: []kio.Filter{&settersutil.FieldSetter{
-					Name:  GcloudProjectNumber,
-					Value: projectNumber, SetBy: "kpt", ResourcesPath: path,
-					OpenAPIPath: filepath.Join(path, "Kptfile")}},
-				Outputs: []kio.Writer{rw},
-			}.Execute()
-			if err != nil {
-				return err
-			}
 		}
 	}
 	return nil
 }
 
-func GetProjectNumberFromProjectID(projectID string) (string, error) {
-	gcloudCmd := exec.Command("gcloud",
-		"projects", "describe", projectID, "--format", "value(projectNumber)")
-	b, err := gcloudCmd.Output()
+// SetV2AutoSetter sets the input auto setter recursively in all the sub-packages of root
+// Sets GcloudProjectNumber as well, if input setter is GcloudProject
+func SetV2AutoSetter(name, value, root string) error {
+	openAPIFileName := "Kptfile"
```

Comment:
Same as above. I assume we have this as a constant somewhere

Reply:
Done.